hedera-node/src/main/java/com/hedera/services/store/schedule/HederaScheduleStore.java

```diff
@@ -216,17 +217,23 @@ private void buildTxBodyMap(Supplier<FCMap<MerkleEntityId, MerkleSchedule>> sche
 		schedulesMap.forEach((key, value) -> txToEntityId.put(Arrays.hashCode(value.transactionBody()), key));
 	}
 
-	public ScheduleID getScheduleIDByTransactionBody(byte[] bodyBytes) {
+	@Override
+	public Optional<ScheduleID> getScheduleID(byte[] bodyBytes, AccountID scheduledTxPayer) {
 		var txHashCode = Arrays.hashCode(bodyBytes);
 
 		if (pendingTxHashCode != null && pendingTxHashCode.equals(txHashCode)) {
```

Comment:
What about the `payer` here?

Reply:
I think that we will have to refactor the `Map<Integer, MerkleEntityId> txToEntityId` to have 2 keys -> the `hashCode` integer and the `payer` account. Otherwise, we might end up in a situation in which we "override" a pending Scheduled TX in `commitCreation` method just because the bodyBytes are the same. We could use private class inside `HederaScheduleStore` for the data structure. 
The class should have the hashCode `Integer` and the `AccountID` scheduledTXPayer